src/commonMain/kotlin/exchange.dydx.abacus/calculator/V2/AdjustIsolatedMarginInputCalculatorV2.kt

```diff
@@ -0,0 +1,207 @@
+                                    modified.amount = amountToRemove * amountPercent
+                                } else {
+                                    modified.amount = null
+                                }
+                            }
+                        }
+                    } else {
+                        modified.amount = null
+                    }
+                }
+            }
+        }
+        return modified
+    }
+
+    private fun getModifiedTransferDelta(
+        isolatedMarginAdjustment: InternalAdjustIsolatedMarginInputState,
+        isParentSubaccount: Boolean,
+    ): Delta {
+        val type = isolatedMarginAdjustment.type ?: IsolatedMarginAdjustmentType.Add
+        val amount = isolatedMarginAdjustment.amount
+
+        when (type) {
+            IsolatedMarginAdjustmentType.Add -> {
+                val multiplier =
+                    if (isParentSubaccount) Numeric.double.NEGATIVE else Numeric.double.POSITIVE
+                val usdcSize = (amount ?: Numeric.double.ZERO) * multiplier
+
+                return Delta(usdcSize = usdcSize)
+            }
+
+            IsolatedMarginAdjustmentType.Remove -> {
+                val multiplier =
+                    if (isParentSubaccount) Numeric.double.POSITIVE else Numeric.double.NEGATIVE
+                val usdcSize = (amount ?: Numeric.double.ZERO) * multiplier
+
+                return Delta(usdcSize = usdcSize)
+            }
+        }
+    }
+
+    private fun summaryForType(
+        parentSubaccount: InternalSubaccountState?,
+        childSubaccount: InternalSubaccountState?,
+    ): AdjustIsolatedMarginInputSummary {
+        val openPositions = childSubaccount?.openPositions
+        val marketId = openPositions?.keys?.firstOrNull()
+        val position = openPositions?.get(marketId)
+
+        return AdjustIsolatedMarginInputSummary(
+            crossFreeCollateral = parentSubaccount?.calculated?.get(CalculationPeriod.current)?.freeCollateral,
+            crossFreeCollateralUpdated = parentSubaccount?.calculated?.get(CalculationPeriod.post)?.freeCollateral,
+            crossMarginUsage = parentSubaccount?.calculated?.get(CalculationPeriod.current)?.marginUsage,
+            crossMarginUsageUpdated = parentSubaccount?.calculated?.get(CalculationPeriod.post)?.marginUsage,
+            positionMargin = position?.calculated?.get(CalculationPeriod.current)?.marginValue,
+            positionMarginUpdated = position?.calculated?.get(CalculationPeriod.post)?.marginValue,
+            positionLeverage = position?.calculated?.get(CalculationPeriod.current)?.leverage,
+            positionLeverageUpdated = position?.calculated?.get(CalculationPeriod.post)?.leverage,
+            liquidationPrice = position?.calculated?.get(CalculationPeriod.current)?.liquidationPrice,
+            liquidationPriceUpdated = position?.calculated?.get(CalculationPeriod.post)?.liquidationPrice,
```

Comment:
super nit, I wonder if this would be easier to read if we had `postPosition` and `currentPosition` variables

Reply:
Can update in the future.... and then there are two calculated fields for subaccount and position, so it gets pretty confusing.